script/check-api-version

```diff
@@ -54,12 +54,15 @@ main()
         exit 0
     fi
 
-    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep -aP '[-+]#define OPENTHREAD_API_VERSION (.+)' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
+    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep '[-+]#define OPENTHREAD_API_VERSION' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
 
-    [[ $(wc -l <"${OT_VERSIONS_FILE}") == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+    line_count="$(wc -l <${OT_VERSIONS_FILE})"
+    line_count="${line_count// /}"
 
-    old_version=$(grep -aoP '(?<=-#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find old version!'
-    new_version=$(grep -aoP '(?<=\+#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find new version!'
+    [[ $line_count == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+
+    old_version=$(grep '\-\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[0-9]..') || die 'Failed to find old version!'
+    new_version=$(grep '\+\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[0-9]..') || die 'Failed to find new version!'
```

Comment:
OpenThread API version is not fixed to 3 digits.

```suggestion
    old_version=$(grep '\-\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[1-9][0-9]*') || die 'Failed to find old version!'
    new_version=$(grep '\+\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[1-9][0-9]*') || die 'Failed to find new version!'
```

Reply:
ok, that looks bette